### PR TITLE
Support only HTTPS for remote upgrade PGP

### PR DESCRIPTION
**Pull request by michalpristas**
## What does this PR do?

Followup to #2246
- restricts URI to be HTTPS
- ~~adds Elastic GPG as a backup~~

!! Manual backport to 7.17

### CODE REVIEW

File: internal/pkg/agent/cmd/upgrade.go [hunk 1]
```diff
@@ -94,3 +100,16 @@ func upgradeCmd(streams *cli.IOStreams, cmd *cobra.Command, args []string) error
...
 	return nil
 }
+
+func checkValidUri(rawURI string) error {
+	uri, err := url.Parse(rawURI)
+	if err != nil {
+		return err
+	}
+
+	if !strings.EqualFold(uri.Scheme, "https") {
```
Comment (michel-laterman) [inline internal/pkg/agent/cmd/upgrade.go:110]: Do we also want to add a similar change for upgrade actions?
https://github.com/elastic/elastic-agent/blob/main/internal/pkg/agent/application/upgrade/artifact/download/verifier.go#L179

Comment (michalpristas) [reply internal/pkg/agent/cmd/upgrade.go:110]: yes we could. good point

**Review by michel-laterman — APPROVED** [review]
lgtm

**Comment by cmacknz** [comment]
Can we sneak in an update to the pgp option in the reference YAML as part of this PR?

https://github.com/elastic/elastic-agent/blob/725b75093ca40a48ff31e60685c52ca1ea9c590b/elastic-agent.reference.yml#L74-L76

**Pull request merged**